nautobot/core/management/commands/generate_performance_test_endpoints.py

```diff
@@ -0,0 +1,228 @@
+        """
+        # If the model class is found, then we know we are dealing with a model related endpoint
+        if model_class:
+            if len(model_class.objects.all()) > 1:
+                # Handle detail view url patterns
+                if "_list" not in view_name and "-list" not in view_name:
+                    # Identify the placeholder for the uuid
+                    replace_string = ""
+                    if "<uuid:pk>" in url_pattern:
+                        replace_string = "<uuid:pk>"
+                    elif "(?P<pk>[/.]+)" in url_pattern:
+                        replace_string = "(?P<pk>[/.]+)"
+
+                    if replace_string:
+                        # Replace the uuid with the actual uuid
+                        first_url_pattern = url_pattern.replace(replace_string, str(model_class.objects.first().pk))
+                        second_url_pattern = url_pattern.replace(replace_string, str(model_class.objects.last().pk))
+                        if view_name not in self.app_name_to_urls["endpoints"]:
+                            self.app_name_to_urls["endpoints"][view_name] = []
+                        self.app_name_to_urls["endpoints"][view_name].append(first_url_pattern)
+                        self.app_name_to_urls["endpoints"][view_name].append(second_url_pattern)
+                # Handle list view url patterns
+                else:
+                    if view_name not in self.app_name_to_urls["endpoints"]:
+                        self.app_name_to_urls["endpoints"][view_name] = []
+                    # One endpoint with default pagination
+                    self.app_name_to_urls["endpoints"][view_name].append(url_pattern)
+                    total_count = len(model_class.objects.all())
+                    page_query_parameter = 5
+                    per_page_query_parameter = total_count // page_query_parameter
+                    if not is_api_endpoint:
+                        query_params = urlencode(
+                            {
+                                "per_page": per_page_query_parameter,
+                                "page": page_query_parameter,
+                            }
+                        )
+                    else:
+                        query_params = urlencode(
+                            {
+                                "limit": per_page_query_parameter,
+                                "offset": per_page_query_parameter * (page_query_parameter - 1),
+                            }
+                        )
+                    # One endpoint with default pagination
+                    self.app_name_to_urls["endpoints"][view_name].append(url_pattern + f"?{query_params}")
+            else:
+                # TODO handle the case where there is no instances of the model is found
+                self.stdout.write(f"No instances of {model_class} found")
```

Comment:
Probably should write to stderr instead - also, this message hits the case where there is 1 instance, not just zero, right?